lib/sipInterconnect.js

```diff
@@ -0,0 +1,155 @@
+  *
+  * @param sessionId The session ID corresponding to the session to which the user will connect.
+  *
+  * @param token The token for the session ID with which the SIP user will use to connect.
+  *
+  * @param sipUri The sip URI the SIP Interconnect feature will dial.
+  *
+  * @param options {Object} An optional options object with the following properties:
+  * <p>
+  * <ul>
+  *   <li>
+  *     <code>headers</code> (Object) &mdash; Custom headers to be added to the SIP INVITE
+  *     request iniated from OpenTok to the Third Party SIP Platform. All headers must start
+  *     with the "X-" prefix, or a Bad Request (400) will be thrown.
+  *   </li>
+  *   <li>
+  *     <code>auth</code> (Object) &mdash; The credentials to be used for HTTP Digest authentication
+  *     in case this is required by the Third Party SIP Platform.
+  *   <ul>
+  *     <li> "username" -- The username to be used in the SIP INVITE.
+  *     <li> "password" -- The password to be used in the SIP INVITE.
+  *   </ul>
+  *   </li>
+  *   <li>
+  *     <code>secure</code> (Boolean) &mdash; Whether the SIP call should be transmitted
+  *     encrypted or not.
+  *   </li>
+  * </ul>
+
+  * @method #dial
+  * @memberof SipInterconnect
+  */
+  this.dial = function() {
+    exports.dial(config, id, token, sipUri, callback);
+  };
+}
+
+var api = function(config, method, path, body, callback) {
+  var rurl = config.apiEndpoint + '/v2/partner/' + config.apiKey + path;
+  if ("defaults" in request) {
+    request = request.defaults(_.pick(config, 'proxy', 'timeout'));
+  }
+  request({
+    url: rurl,
+    method: method,
+    headers: {
+      'X-TB-PARTNER-AUTH': config.apiKey + ':' + config.apiSecret,
+      'User-Agent': 'OpenTok-Node-SDK/' + pkg.version +
+        (config.uaAddendum ? ' ' + config.uaAddendum : '')
+    },
+    json: body
+  }, callback);
+};
+
+exports.dial = function(config, sessionId, token, sipUri, options, callback) {
+  if(typeof options == 'function') {
+    callback = options;
+    options = {};
+  }
+  if(typeof callback != 'function') {
+    throw(new errors.ArgumentError('No callback given to dial'));
+  }
+  if(sessionId == null || sessionId.length === 0) {
+    callback(new errors.ArgumentError('No session ID given'));
+    return;
+  }
+  var body = {
+	sessionId: sessionId,
+	token: token,
+	sip: {
+		uri: sipUri
+	}
+  };
+  if (_.isObject(options.headers) && !_.isArray(options.headers)) {
+  	body.sip.headers = options.headers;
+  };
+  if (_.isObject(options.auth) && !_.isArray(options.auth)) {
+  	body.sip.auth = options.auth;
+  };
+  if (!!options.secure) {
+  	body.sip.secure = options.secure;
+  }
+  api(config, 'POST', '/call', body,
+  function(err, response, body) {
+  	console.log([response.statusCode, body]);
+    if(err) {
+      callback(err);
+    } else if(response.statusCode != 200) {
+      if(response && response.statusCode == 404) {
```

Comment:
find out if these are real response codes. and look to see if there are any that arent covered here